src/slicegpt/slicing_scheduler.py

```diff
@@ -259,3 +301,38 @@ def linear_sparsity_func(location: float) -> float:
             round_interval=round_interval,
             do_slice_head=do_slice_head,
         )
+
+
+class ExplainedVarianceSlicingScheduler(ForwardSlicingScheduler):
+    """A slicing scheduler that applies sparsity based on the explained variance from the PCA."""
+
+    def __init__(
+        self,
+        *,
+        uev_threshold: float,
+        round_interval: int = 1,
+        do_slice_head: bool = False,
+    ):
+        super().__init__(do_slice_head=do_slice_head)
+        self.uev_threshold: float = uev_threshold
+        self.round_interval: int = round_interval
+
+    def _get_layer_dimension(self, eigen_vals: list[float], plot: bool = False) -> int:
+        eigen_vals = np.array(eigen_vals)
+        cum_var = np.cumsum(np.array(eigen_vals)) / np.sum(eigen_vals)
+        dim = np.argmax(cum_var > 1 - self.uev_threshold)
+        dim -= dim % self.round_interval
+        dim = int(dim)
```

Comment:
Is it worth having the `round_interval` parameter passed to slicing schedulers in general, to ensure they always return dims that are rounded accordingly? I guess the constant and function schedulers could ask for constants/functions that take the rounding interval into account already, but what about this scheduler?